.github/index.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html lang='en'>
+<head>
+  <meta charset='UTF-8'>
+  <title>My Photohub</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <style>
+    body { margin: 0; background: #333; }
+    header { 
+      padding: .5vw;
+      font-size: 0;
+      display: -ms-flexbox;
+      -ms-flex-wrap: wrap;
+      -ms-flex-direction: column;
+      -webkit-flex-flow: row wrap; 
+      flex-flow: row wrap; 
+      display: -webkit-box;
+      display: flex;
+    }
+    header div { 
+      -webkit-box-flex: auto;
+      -ms-flex: auto;
+      flex: auto; 
+      width: 200px; 
+      margin: .5vw; 
+    }
+    header div img { 
+      width: 100%; 
+      height: auto; 
+    }
+    @media screen and (max-width: 400px) {
+      header div { margin: 0; }
+      header { padding: 0; }
+    }
+  </style>
+<body>
+  <header id="gallery">
+  </header>
+  <script>
+    const addImg = (url) => {
+      if (!url) return
+      const gallery = document.querySelector('#gallery')
+      const img = document.createElement("img")
+      img.src = `./optimized/${url}`
+      const division = document.createElement("div")
+      division.appendChild(img)
+      gallery.appendChild(division)
+    }
+
+    fetch("./filelist.txt")
```

Comment:
This technique is fine as a first pass, but we don't need to dynamically load the file list and build the DOM, since we already have a build step (i.e., GitHub Actions) and can generate the static HTML with all images before we commit the file to git.  This will give a much better experience to the user, since the initial load will be faster, and the images can be properly positioned and sized (i.e., our build step can also figure this out, and add intrinsic sizes).

If you want to leave this optimization for later, that's fine.  File a follow-up issue.